### PR TITLE
lib: bin: lwm2m_carrier: update to version 0.8.2

### PR DESCRIPTION
**Pull request by lemrey**
Update library to version 0.8.2.

### CODE REVIEW

File: lib/bin/lwm2m_carrier/include/lwm2m_carrier.h [hunk 1]
```diff
@@ -36,13 +36,40 @@ typedef struct {
...
+} lwm2m_carrier_event_deferred_t;
+
 /**
  * @brief LWM2M carrier library event error codes.
  */
 #define LWM2M_CARRIER_ERROR_NO_ERROR        0 /**< No error. */
 #define LWM2M_CARRIER_ERROR_CONNECT_FAIL    1 /**< Failed to connect to the LTE network. */
 #define LWM2M_CARRIER_ERROR_DISCONNECT_FAIL 2 /**< Failed to disconnect from the LTE network. */
 #define LWM2M_CARRIER_ERROR_BOOTSTRAP       3 /**< LWM2M carrier bootstrap failed. */
+#define LWM2M_CARRIER_ERROR_FOTA_PKG        4 /**< Update package rejected. */
```
Comment (stig-bjorlykke) [inline lib/bin/lwm2m_carrier/include/lwm2m_carrier.h:68]: `samples/nrf9160/lwm2m_carrier/src/main.c` must be updated with new error codes.

File: lib/bin/lwm2m_carrier/include/lwm2m_carrier.h [hunk 2]
```diff
@@ -36,13 +36,40 @@ typedef struct {
...
+
+/*
+ * @brief LwM2M carrier library deferred event structure.
+ */
+typedef struct {
+	/** Deferred event reason. */
+	uint32_t reason;
+	/** Time before operation is resumed (seconds). */
+	int32_t timeout;
+} lwm2m_carrier_event_deferred_t;
```
Comment (stig-bjorlykke) [inline lib/bin/lwm2m_carrier/include/lwm2m_carrier.h:59]: `samples/nrf9160/lwm2m_carrier/src/main.c` should be updated to show `reason` and `timeout`.

Comment (lemrey) [reply lib/bin/lwm2m_carrier/include/lwm2m_carrier.h:59]: Done

File: lib/bin/lwm2m_carrier/include/lwm2m_carrier.h (hunk 1 above)
Comment (lemrey) [reply lib/bin/lwm2m_carrier/include/lwm2m_carrier.h:68]: Done

File: samples/nrf9160/lwm2m_carrier/src/main.c [hunk 3]
```diff
@@ -57,9 +67,13 @@ int lwm2m_carrier_event_handler(const lwm2m_carrier_event_t *event)
 	case LWM2M_CARRIER_EVENT_READY:
 		printk("LWM2M_CARRIER_EVENT_READY\n");
 		break;
-	case LWM2M_CARRIER_EVENT_DEFERRED:
+	case LWM2M_CARRIER_EVENT_DEFERRED: {
+		lwm2m_carrier_event_deferred_t *devt = event->data;
 		printk("LWM2M_CARRIER_EVENT_DEFERRED\n");
+		printk("Reason %u, timeout %d seconds\n",
```
Comment (stig-bjorlykke) [inline samples/nrf9160/lwm2m_carrier/src/main.c:73]: We print a string representation of the error so we should probably also print a string representation of reason here.

**Comment by metc** [comment]
LGTM thank-you @lemrey

**Comment by lemrey** [comment]
Rebased

**Comment by lemrey** [comment]
Updated manifest

**Pull request merged**